src/components/transport_manager/src/bluetooth/bluetooth_connection_factory.cc

```diff
@@ -57,12 +58,20 @@ TransportAdapter::Error BluetoothConnectionFactory::CreateConnection(
   const DeviceUID& device_uid, const ApplicationHandle& app_handle) {
   LOG4CXX_TRACE(logger_, "enter. device_uid: " << &device_uid << ", app_handle: " <<
                 &app_handle);
-  BluetoothSocketConnection* connection(
-    new BluetoothSocketConnection(device_uid, app_handle, controller_));
-  TransportAdapter::Error error = connection->Start();
-  if (TransportAdapter::OK != error) {
-    LOG4CXX_ERROR(logger_, "connection::Start() failed");
-    delete connection;
+  TransportAdapter::Error error = TransportAdapter::ALREADY_EXISTS;
+  const ConnectionMapIndex connection_index(device_uid, app_handle);
+  if (connections_.end() == connections_.find(connection_index))
+  {
+    ConnectionSPtr connection(
+        new BluetoothSocketConnection(device_uid, app_handle, controller_));
+    BluetoothSocketConnection* bt_connection =
+        static_cast<BluetoothSocketConnection*>(connection.get());
```

Comment:
I hope we have a casting inside of `SharesPointer` and the following code shall works well
```
BluetoothSocketConnectionSptr bt_connection = MakeShared(...);
bt_connection-> start();
connection_[index] = bt_connection;
```